integration/tests/observer/execution_state_sync_test.go

```diff
@@ -0,0 +1,208 @@
+		testnet.NewNodeConfig(flow.RoleCollection, testnet.WithLogLevel(zerolog.FatalLevel)),
+		testnet.NewNodeConfig(flow.RoleExecution, testnet.WithLogLevel(zerolog.FatalLevel)),
+		testnet.NewNodeConfig(flow.RoleExecution, testnet.WithLogLevel(zerolog.FatalLevel)),
+		testnet.NewNodeConfig(flow.RoleConsensus, consensusConfigs...),
+		testnet.NewNodeConfig(flow.RoleConsensus, consensusConfigs...),
+		testnet.NewNodeConfig(flow.RoleConsensus, consensusConfigs...),
+		testnet.NewNodeConfig(flow.RoleVerification, testnet.WithLogLevel(zerolog.FatalLevel)),
+		bridgeANConfig,
+		ghostNode,
+		// TODO: add observer
+	}
+
+	conf := testnet.NewNetworkConfig("execution state sync test", net)
+	s.net = testnet.PrepareFlowNetwork(s.T(), conf, flow.Localnet)
+}
+
+// TestHappyPath tests that Execution Nodes generate execution data, and Access Nodes are able to
+// successfully sync the data
+func (s *ExecutionStateSyncSuite) TestHappyPath() {
+	// Let the network run for this many blocks
+	runBlocks := uint64(20)
+
+	// We will check that execution data was downloaded for this many blocks
+	// It has to be less than runBlocks since it's not possible to see which height the AN stopped
+	// downloading execution data for
+	checkBlocks := runBlocks / 2
+
+	// get the first block height
+	blockA := s.BlockState.WaitForHighestFinalizedProgress(s.T())
+	s.T().Logf("got block height %v ID %v", blockA.Header.Height, blockA.Header.ID())
+
+	// wait for the requested number of sealed blocks, then pause the network so we can inspect the dbs
+	s.BlockState.WaitForSealed(s.T(), blockA.Header.Height+runBlocks)
+	s.net.StopContainers()
+
+	// start an execution data service using the Access Node's execution data db
+	an := s.net.ContainerByID(s.bridgeID)
+	eds, ctx := s.nodeExecutionDataService(an)
+
+	// setup storage objects needed to get the execution data id
+	db, err := an.DB()
+	require.NoError(s.T(), err, "could not open db")
+
+	metrics := metrics.NewNoopCollector()
+	headers := storage.NewHeaders(metrics, db)
+	results := storage.NewExecutionResults(metrics, db)
+
+	// Loop through checkBlocks and verify the execution data was downloaded correctly
+	for i := blockA.Header.Height; i <= blockA.Header.Height+checkBlocks; i++ {
+		header, err := headers.ByHeight(i)
+		require.NoError(s.T(), err, "could not get header")
+
+		result, err := results.ByBlockID(header.ID())
+		require.NoError(s.T(), err, "could not get sealed result")
+
+		s.T().Logf("getting execution data for height %d, block %s, execution_data %s", header.Height, header.ID(), result.ExecutionDataID)
+
+		ed, err := eds.Get(ctx, result.ExecutionDataID)
+		assert.NoError(s.T(), err, "could not get execution data for height %v", i)
+
+		s.T().Logf("got execution data for height %d", i)
+		assert.Equal(s.T(), header.ID(), ed.BlockID)
+	}
+}
+
+func (s *ExecutionStateSyncSuite) nodeExecutionDataService(node *testnet.Container) (state_synchronization.ExecutionDataService, irrecoverable.SignalerContext) {
```

Comment:
rename this to something like `getTestExecutionDataServiceNode`

Reply:
Fixed in: https://github.com/GetElastech/flow-go/commit/5078c8c76b61ec4fb9acf3caefecb365c358b1ba